### PR TITLE
Migrate battery and temp_info to the new lua module syntax.

### PR DESCRIPTION
**Pull request by francois2metz**
See #36 

I tried to keep the compatibility. I don't know how to test that.

### CODE REVIEW

**Comment by hoelzro** [comment]
Looks good at first glance - I'll give it a more thorough look later!

**Comment by hoelzro** [comment]
So one thing I'm seeing is a lot of removal of things like `local assert = assert` - those weren't just here to make sure the code could refer to them within a `module`, but also for performance reasons (eg. see `battery.backends` - it doesn't use `module` and still has those).  I'll still merge the code, but I'll restore those later. Just a heads up if you decide to do future work on Lua 5.3 compatability!

File: battery/init.lua [hunk 1]
```diff
@@ -152,13 +138,14 @@ widget:buttons(awful.util.table.join(
 -- XXX FIXME
 lib.hooks.timer.register(1, 300, update)
 
-setmetatable(_M, { __call = function ()
+setmetatable(battery, { __call = function ()
   if not backend then
-    backend = backends.get(_M.preferred_backend)
+     backend = backends.get()
```
Comment (hoelzro) [inline battery/init.lua]: We're going to want to continue using `preferred_backend` so that users can specify a preferred battery backend.

**Comment by hoelzro** [comment]
Also, a tip going forward porting things to Lua 5.3 (not something I see in your changes, just something I'd like to bring attention to): we don't want to contaminate the global state.  You can check to see creation of globals with `luac -p -l $lua_file | grep SETTABUP | grep _ENV` - if there's no output, there are no accidental globals =)

File: battery/init.lua (hunk 1 above)
Comment (francois2metz) [reply battery/init.lua]: Fixed. Thanks.

**Comment by francois2metz** [comment]
I restored the local variables, fixed the `preferred_backend` and updated the readme.

**Comment by hoelzro** [comment]
Thanks for your patience - this looks good to me! I'm going to throw in a small warning so that users can transition their configs to the new way of doing things (now that you you need to do `local temp_info = require 'obvious.temp_info'; temp_info()` instead of just `require 'obvious.temp_info; obvious.temp_info()`)

**Pull request merged**